### PR TITLE
Switching to using the new Sdk

### PR DESCRIPTION
**Pull request by bettinaheim**
This PR removes the code gen app and switches to instead using the Microsoft Quantum Sdk. 
It also splits out a separate package for Microsoft.Quantum.QSharp.Core and Microsoft.Quantum.Runtime.Core.

### CODE REVIEW

**Comment by bettinaheim** [comment]
I think the currently issue might be that the way the current Quantum Sdk version is set up, we are somehow loosing the .pdb files from references (they should be copied to the output folder).

**Comment by bettinaheim** [comment]
The latest two build failures do not seem to be related to any changes that I did. Seems like a weird issue that is hopefully transient (memory access violation error).

**Comment by bettinaheim** [comment]
It looks like switching form .NET Core Sdk 3.0.100 to 3.1.100 causes an AccessViolationException in the Borrowing test (under Microsoft.Quantum.Simulation.Simulators.Tests, CoreTests). @avasch01, @anpaz-msft - any ideas?

**Comment by avasch01** [comment]
So what is the issue and how can I repro it?

**Comment by bettinaheim** [comment]
> So what is the issue and how can I repro it?
> 
> From: bettinaheim <notifications@github.com>
> Sent: Wednesday, December 25, 2019 10:43 PM
> To: microsoft/qsharp-runtime <qsharp-runtime@noreply.github.com>
> Cc: Alexander Vaschillo <alexva@microsoft.com>; Mention <mention@noreply.github.com>
> Subject: Re: [microsoft/qsharp-runtime] Switching to using the new Sdk (#113)
> 
> 
> It looks like switching form .NET Core Sdk 3.0.100 to 3.1.100 causes an AccessViolationException in the Borrowing test (under Microsoft.Quantum.Simulation.Simulators.Tests, CoreTests). @avasch01<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Favasch01&data=02%7C01%7Calexva%40microsoft.com%7Ca3d40e18d55d46e4a10f08d789ced0bb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637129393667810325&sdata=OyddeO5542H8prl0LXfV%2FFN%2BScfeAF9dTwTYtehI4LQ%3D&reserved=0>, @anpaz-msft<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fanpaz-msft&data=02%7C01%7Calexva%40microsoft.com%7Ca3d40e18d55d46e4a10f08d789ced0bb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637129393667810325&sdata=zD5ujwF1BKTaJlVuajvKCp%2B6%2FLFxRWau5nXtnjiVrbU%3D&reserved=0> - any ideas?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmicrosoft%2Fqsharp-runtime%2Fpull%2F113%3Femail_source%3Dnotifications%26email_token%3DAIRQIC3RPICIF2UX6ANJ7MDQ2RG6JA5CNFSM4J5TDRH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHVBHNA%23issuecomment-568988596&data=02%7C01%7Calexva%40microsoft.com%7Ca3d40e18d55d46e4a10f08d789ced0bb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637129393667820282&sdata=%2Foa2C9b2OgN8GiaPEkCA%2FY6Pfg7l5vDe5SX%2B3juewyM%3D&reserved=0>, or unsubscribe<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAIRQIC3GKBA3TVTFX5BG7I3Q2RG6JANCNFSM4J5TDRHQ&data=02%7C01%7Calexva%40microsoft.com%7Ca3d40e18d55d46e4a10f08d789ced0bb%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637129393667820282&sdata=EQiezJZrSucteVM6UGwhxKFy%2BC4vCCNmzj3rgqVWdfY%3D&reserved=0>.

I filed it here: https://github.com/microsoft/qsharp-runtime/issues/116
Basically, download the latest .NET Core Sdk, and try to run the Borrowing test on master.

**Comment by bettinaheim** [comment]
> 🕐

I have no idea what that mean... :)

File: build/pack.ps1 [hunk 1]
```diff
@@ -34,11 +34,31 @@ function Pack-One() {
     }
 }
 
+function Pack-One-Dotnet() {
```
Comment (cgranade) [inline build/pack.ps1]: Minor PowerShell style note, but cmdlet names generally follow the form `Verb-Noun`, where both `Verb` and `Noun` follow PascalCase.
```suggestion
function Pack-OneDotnet() {
```

Comment (bettinaheim) [reply build/pack.ps1]: renamed

File: src/Simulation/Simulators/FindNuspecReferences.ps1 [hunk 2]
```diff
@@ -31,42 +31,73 @@ if (Test-Path $target) {
...
-    # Recursively check on project references:
     $projectDependency = $csproj.Project.ItemGroup.ProjectReference | Where-Object { $null -ne $_ }
     $projectDependency | ForEach-Object {
+        $id = $_.Include
+        Write-Host "Detected project dependencies: $id"
+    }
+
+    # Assume there is a package for project references that are not tagged as to be included in the simulator package:
+    $projectDependency | Where-Object {$_.IncludeInSimulatorPackage -ne 'true' -and $_.IsQscReference -ne 'true'} | ForEach-Object {
+        Add-PackageReference-IfNew $_ $dep
```
Comment (cgranade) [inline src/Simulation/Simulators/FindNuspecReferences.ps1]: There's two parameters passed here, but `Add-PackageReference-IfNew` only declares one formal parameter?

File: build/pack.ps1 [hunk 3]
```diff
@@ -34,11 +34,31 @@ function Pack-One() {
     }
 }
 
+function Pack-Dotnet() {
+    Param($project, $option1="", $option2="", $option3="")
```
Comment (cgranade) [inline build/pack.ps1]: You can use the `$Args` automatic variable to capture extra parameters:

```pwsh
PS> function Foo { param($x); Write-Host $Args; }
PS> Foo 3 4 -b 5
4 -b 5
```

File: src/Simulation/Simulators/FindNuspecReferences.ps1 (hunk 2 above)
Comment (bettinaheim) [reply src/Simulation/Simulators/FindNuspecReferences.ps1]: I pretty much copied the signature of the method below - either one is fine with me.

File: build/pack.ps1 (hunk 3 above)
Comment (bettinaheim) [reply build/pack.ps1]: Sure, again, copy pasted (occasionally even I do that ;) -  for the sake of sticking with what is already there).

File: src/Simulation/Simulators/FindNuspecReferences.ps1 (hunk 2 above)
Comment (bettinaheim) [reply src/Simulation/Simulators/FindNuspecReferences.ps1]: removed it

Comment (cgranade) [reply src/Simulation/Simulators/FindNuspecReferences.ps1]: Ah, I thought it was new from the way it showed up in diffs. My apologies!

**Pull request merged**